extends Control

# åœ¨å…¶ä»–è„šæœ¬ä¸­ä½¿ç”¨
@onready var animated_text = $AnimatedText

func _ready():
    animated_text.connect("text_appeared", _on_text_appeared)
    animated_text.connect("text_cleared", _on_text_cleared)
    
    # è®¾ç½®æ–‡æœ¬
    animated_text.set_text("Hello World! yes,go ğŸ‘‹ ä½ å¥½ ä½ å¥½")
    animated_text.enable_shake = false
    # animated_text.enable_wave = false

    $BtnTest.pressed.connect(_changed_text)
    $BtnRainbow.pressed.connect(setup_custom_rainbow)

    # å¼€å¯å½©è™¹æ•ˆæœ
    animated_text.enable_rainbow = true
    animated_text.rainbow_speed = 1.0  # è°ƒæ•´é¢œè‰²å˜åŒ–é€Ÿåº¦
    animated_text.rainbow_phase = 0.1  # è°ƒæ•´å­—ç¬¦é—´çš„é¢œè‰²å·®å¼‚

func _on_text_appeared():
    print("æ–‡æœ¬å®Œå…¨æ˜¾ç¤ºå‡ºæ¥äº†ï¼")

func _on_text_cleared():
    print("æ–‡æœ¬å®Œå…¨æ¸…é™¤äº†ï¼")

func _changed_text():
    var txt = Rnd.pick([
            "Hello Worl! ",
            "Pello Worl! ",
            "Helkg Dola!",
            "Hello World! yes",
            "Hello World! yes,go ğŸ‘‹ ä½ å¥½ ä½ å¥½",
            "ä½ å¥½ ä½ å¥½",
        ])
    animated_text.set_text(txt)

# é«˜çº§ä½¿ç”¨
func setup_custom_rainbow():
    animated_text.set_text("Custom Rainbow!")
    for i in animated_text.get_children():
        if i is AnimatedText.CharNode:
            i.enable_rainbow = true
            i.rainbow_speed = 2.0
            i.rainbow_saturation = 0.9  # æ›´é²œè‰³çš„é¢œè‰²
            i.rainbow_value = 1.0       # æœ€å¤§äº®åº¦
